org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/JSEditorAssociationOverride.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Victor Rubezhny (Red Hat Inc.) - initial implementation
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.ui.IEditorDescriptor;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.ide.IEditorAssociationOverride;
+import org.eclipse.ui.internal.genericeditor.ExtensionBasedTextEditor;
+import org.eclipse.ui.internal.genericeditor.GenericEditorWithContentTypeIcon;
+
+public class JSEditorAssociationOverride implements IEditorAssociationOverride {
+
+	private Map<String, IEditorDescriptor> descriptorMap = new HashMap<>();
+
+	@Override
+	public IEditorDescriptor[] overrideEditors(IEditorInput editorInput, IContentType contentType,
+			IEditorDescriptor[] editorDescriptors) {
+		return editorInput != null ? overrideEditors(editorInput.getName(), contentType, editorDescriptors)
+				: editorDescriptors;
+	}
+
+	@Override
+	public IEditorDescriptor[] overrideEditors(String fileName, IContentType contentType,
+			IEditorDescriptor[] editorDescriptors) {
+		return Arrays.stream(editorDescriptors).map(descriptor -> getEditorDescriptorForFile(descriptor, fileName)).toArray(size -> new IEditorDescriptor[size]);
```

Comment:
you can replace `size -> new IEditorDescriptor[size])` by `IEditorDescriptor[]::new`

Reply:
This code is is dropped in favour of overriding the `overrideDefaultEditor(...)` methods